src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -98,11 +107,63 @@ protected SSTableWriter(Builder<?, ?> builder, LifecycleNewTracker lifecycleNewT
         this.isTransient = builder.isTransientSSTable();
         this.metadataCollector = builder.getMetadataCollector();
         this.header = builder.getSerializationHeader();
-        this.observers = builder.getFlushObservers();
         this.mmappedRegionsCache = builder.getMmappedRegionsCache();
         this.lifecycleNewTracker = lifecycleNewTracker;
 
+        ArrayList<SSTableFlushObserver> observers = new ArrayList<>();
+        this.observers = Collections.unmodifiableList(observers);
```

Comment:
nit: Couldn't these 2 lines move down to above the `try` below that populates the observers. I find reading the code that flicking up and down is painful.